rules/aws_resource_missing_tags_test.go

```diff
@@ -259,6 +259,127 @@ rule "aws_resource_missing_tags" {
 				},
 			},
 		},
+		{
+			Name: "Default tags multiple providers",
+			Content: `
+provider "aws" {
+  default_tags {
+    tags = {
+      "Fooz": "Barz"
+      "Bazz": "Quxz"
+    }
+  }
+}
+
+provider "aws" {
+  alias = "foo"
+  default_tags {
+    tags = {
+      "Bazz": "Quxz"
+      "Fooz": "Barz"
+    }
+  }
+}
+
+resource "aws_instance" "ec2_instance" {
+  instance_type = "t2.micro"
+}
+
+resource "aws_instance" "ec2_instance_alias" {
+  provider = "aws.foo"
+  instance_type = "t2.micro"
+}`,
+			Config: `
+rule "aws_resource_missing_tags" {
+  enabled = true
+  tags = ["Bazz", "Fooz"]
+}`,
+			Expected: helper.Issues{},
+		},
+		{
+			Name: "Default Tags Are to Be overriden by resource specific tags",
+			Content: `
+provider "aws" {
+  default_tags {
+    tags = {
+      "Foo": "Bar"
+    }
+  }
+}
+
+resource "aws_instance" "ec2_instance" {
+  instance_type = "t2.micro"
+  tags = {
+    "Foo": "Bazz"
+  }
+}`,
+			Config: `
+rule "aws_resource_missing_tags" {
+  enabled = true
+  tags = ["Foo"]
+}`,
+			Expected: helper.Issues{},
+		},
+		{
+			Name: "Resource specific tags are not needed if default tags are placed",
+			Content: `
+provider "aws" {
+  default_tags {
+    tags = {
+      "Foo": "Bar"
+    }
+  }
+}
+
+resource "aws_instance" "ec2_instance" {
+  instance_type = "t2.micro"
+}`,
+			Config: `
+rule "aws_resource_missing_tags" {
+  enabled = true
+  tags = ["Foo"]
+}`,
+			Expected: helper.Issues{},
+		},
+		{
+			Name: "Resource tags in combination with provider level tags",
+			Content: `
+provider "aws" {
+  default_tags {
+    tags = {
+      "Foo": "Bar"
+    }
+  }
+}
+
+resource "aws_instance" "ec2_instance_fail" {
+  instance_type = "t2.micro"
+}
+
+
+resource "aws_instance" "ec2_instance" {
+  instance_type = "t2.micro"
+  tags = {
+    "Bazz": "Quazz"
+  }
+}`,
+			Config: `
+rule "aws_resource_missing_tags" {
+  enabled = true
+  tags = ["Foo", "Bazz"]
+}`,
+			Expected: helper.Issues{
+				{
+					Rule:    NewAwsResourceMissingTagsRule(),
+					Message: "The resource is missing the following tags: \"Bazz\".",
+					Range: hcl.Range{
+						Filename: "module.tf",
+						Start:    hcl.Pos{Line: 10, Column: 1},
+						End:      hcl.Pos{Line: 10, Column: 44},
+					},
+				},
+			},
+		},
```

Comment:
It would be nice to have a test for the alias.

Reply:
Oops, sorry, I missed it.

Since CI does not fail for https://github.com/terraform-linters/tflint-ruleset-aws/pull/489#discussion_r1200585039, I thought that there was no test for alias.
Perhaps it would be better to have a test against non-evaluable aliases like `aws.west` (not `"aws.west"`).

Reply:
Done, added an specific test for this case in [here](https://github.com/JorgeReus/tflint-ruleset-aws/blob/feat/default_tags/rules/aws_resource_missing_tags_test.go#L405)